src/framework/execution/dependencyGraph.ts

```diff
@@ -104,16 +104,15 @@ export function executeStepDependencyGraph(
    * NOTE: the untouched input graph is used for determining
    * results since the working graph gets mutated.
    */
-  function stepHasDependencyFailure(step: IntegrationStep) {
+  function stepHasDependencyFailure(stepId: string) {
```

Comment:
Was it better to use a `string` for argument instead of `IntegrationStep`? It seems like there is less safety for making this a string. Also, we should consider using `Opaque` type when you string identifiers like this. It would allow you to do something like this:

```
export type Opaque<K, T> = T & { __TYPE__: K };
export type StepId = Opaque<'StepId',string>;
```

I started using this `Opaque` type in other projects. See https://evertpot.com/opaque-ts-types/

Reply:
`IntegrationStep` has been completely replaced with `Step` in this file. Most functions in this file only require the step ID, but were taking the entire step. The new type generics make this very verbose and I was trying to simplify it but only passing what the function actually needs. It's more pure and easier to maintain this way.